editor/editor_properties.cpp

```diff
@@ -3693,12 +3750,23 @@ EditorPropertyNodePath::EditorPropertyNodePath() {
 	SET_DRAG_FORWARDING_CD(assign, EditorPropertyNodePath);
 	hbc->add_child(assign);
 
-	clear = memnew(Button);
-	clear->set_flat(true);
-	clear->connect("pressed", callable_mp(this, &EditorPropertyNodePath::_node_clear));
-	hbc->add_child(clear);
-
-	scene_tree = nullptr; //do not allocate unnecessarily
+	menu = memnew(MenuButton);
+	menu->set_flat(true);
+	menu->connect(SNAME("about_to_popup"), callable_mp(this, &EditorPropertyNodePath::_update_menu));
+	hbc->add_child(menu);
+
+	menu->get_popup()->add_item(TTR("Clear"), ACTION_CLEAR);
```

Comment:
Just curious: Why not take a reference/pointer to the popup and then `popup->add_item(x)`? I'm seeing such pattern (repeating getters or whatever on multiple lines) quite often in Godot's code base.

Reply:
idk, I just haven't thought about that. It's not like it's done consistently.